ecosystem/indexer-grpc/indexer-grpc-data-service/src/service.rs

```diff
@@ -422,24 +479,59 @@ async fn data_fetch(
     cache_operator: &mut CacheOperator<redis::aio::ConnectionManager>,
     file_store_operator: &dyn FileStoreOperator,
 ) -> anyhow::Result<TransactionsDataStatus> {
+    let start_time = std::time::Instant::now();
     let batch_get_result = cache_operator
         .batch_get_encoded_proto_data(starting_version)
         .await;
 
     match batch_get_result {
         // Data is not ready yet in the cache.
         Ok(CacheBatchGetStatus::NotReady) => Ok(TransactionsDataStatus::AheadOfCache),
-        Ok(CacheBatchGetStatus::Ok(transactions)) => Ok(TransactionsDataStatus::Success(
+        Ok(CacheBatchGetStatus::Ok(transactions)) => {
+            let size_in_bytes = transactions
+                .iter()
+                .map(|transaction| transaction.len())
+                .sum::<usize>();
+            let num_of_transactions = transactions.len();
+            let duration_in_secs = start_time.elapsed().as_secs_f64();
+            info!(
+                start_version = starting_version,
+                end_version = starting_version + num_of_transactions as u64 - 1,
+                num_of_transactions = transactions.len(),
+                size_in_bytes = size_in_bytes,
+                duration_in_secs = duration_in_secs,
+                tps = num_of_transactions as f64 / duration_in_secs,
+                bytes_per_sec = size_in_bytes as f64 / duration_in_secs,
+                "[Indexer Data] Data fetched from cache."
+            );
+            Ok(TransactionsDataStatus::Success(
             build_protobuf_encoded_transaction_wrappers(transactions, starting_version),
-        )),
+        ))},
         Ok(CacheBatchGetStatus::EvictedFromCache) => {
             // Data is evicted from the cache. Fetch from file store.
             let file_store_batch_get_result =
                 file_store_operator.get_transactions(starting_version).await;
             match file_store_batch_get_result {
-                Ok(transactions) => Ok(TransactionsDataStatus::Success(
+                Ok(transactions) => {
+                    let size_in_bytes = transactions
+                    .iter()
+                    .map(|transaction| transaction.len())
+                    .sum::<usize>();
+                let num_of_transactions = transactions.len();
+                let duration_in_secs = start_time.elapsed().as_secs_f64();
+                info!(
+                    start_version = starting_version,
+                    end_version = starting_version + num_of_transactions as u64 - 1,
+                    num_of_transactions = transactions.len(),
+                    size_in_bytes = size_in_bytes,
+                    duration_in_secs = duration_in_secs,
+                    tps = num_of_transactions as f64 / duration_in_secs,
+                    bytes_per_sec = size_in_bytes as f64 / duration_in_secs,
+                    "[Indexer Data] Data fetched from cache."
```

Comment:
This should say file store, also add service_type = file_store

Reply:
done.